plugins/actsvg/include/detray/plugins/actsvg/surface_svg_converter.hpp

```diff
@@ -0,0 +1,133 @@
+#include <math.h>
+
+#include <type_traits>
+#include <vector>
+
+// Project include(s)
+#include "actsvg/meta.hpp"
+#include "actsvg/proto/surface.hpp"
+#include "detray/definitions/units.hpp"
+#include "detray/geometry/surface.hpp"
+
+namespace detray::actsvg_visualization {
+
+using point3 = std::array<actsvg::scalar, 3>;
+using point3_container = std::vector<point3>;
+using proto_surface = actsvg::proto::surface<point3_container>;
+
+template <typename matrix_t>
+inline std::array<detray::scalar, 3> rotation_matrix_to_euler_angles(
+    const matrix_t& mat) {
+    float a = std::sqrt(mat[0][0] * mat[0][0] + mat[1][0] * mat[1][0]);
+    // Checking if it is singular.
+    if (a < 1e-6)
+        return {std::atan2(-mat[1][2], mat[1][1]), std::atan2(-mat[2][0], a),
+                0};
+
+    return {std::atan2(mat[2][1], mat[2][2]), std::atan2(-mat[2][0], a),
+            std::atan2(mat[1][0], mat[0][0])};
+}
+
+template <typename transform_t>
+inline auto convert_transform(const transform_t& d_transform) {
+    auto translation = d_transform.translation();
+    auto euler_angles =
+        rotation_matrix_to_euler_angles<>(d_transform.rotation());
+
+    // TODO: skew and scale
+
+    auto ret = actsvg::style::transform();
+    constexpr auto rad_to_deg = 180.0 / 3.14;
+    ret._tr = {static_cast<actsvg::scalar>(translation[0]),
+               static_cast<actsvg::scalar>(translation[1])};
+    ret._rot = {static_cast<actsvg::scalar>(euler_angles[2] * rad_to_deg),
+                static_cast<actsvg::scalar>(euler_angles[1] * rad_to_deg),
+                static_cast<actsvg::scalar>(euler_angles[0] * rad_to_deg)};
+
+    return ret;
+}
+
+template <class mask_t>
+inline proto_surface convert_mask(const mask_t& m) {
+    proto_surface p_surface;
+
+    const auto& shape = m.get_shape();
+
+    using shape_t = typename mask_t::shape;
+    constexpr auto is_annulus2D = std::is_same_v<shape_t, detray::annulus2D<>>;
```

Comment:
You will probably have to use the ```std::is_same_v<>``` expressions in the ```if```` directly, in order for the CI to stop complaining about unused variables